### PR TITLE
Implement a more useful `ensureMinCoinTxOut`

### LINKED ISSUE #3247: Add `ensureMinCoinTxOut`
I think it is also possible that someone would like to just ensure that there TxOut satisfies the minimum output condition, rather than set it to the minimum possible value. It would be good to add a function that is very similar to `setMinCoinTxOut`, but is a bit more lenient.

### PR DESCRIPTION
**Pull request by lehins**
# Description

Fixes #3247 

Also switches `cardano-ledger-api` test suite to `hspec`

# Checklist

- [x] Commit sequence broadly makes sense
- [x] Commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] Any changes are noted in the [changelog](https://github.com/input-output-hk/cardano-ledger/blob/master/CHANGELOG.md)
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (which can be run with `scripts/fourmolize.sh`
- [x] Self-reviewed the diff

### CODE REVIEW

File: libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Out.hs [hunk 1]
```diff
@@ -43,45 +48,86 @@ module Cardano.Ledger.Api.Tx.Out (
...
+  TxOut era ->
+  TxOut era
+setMinCoinTxOutInternal f pp = go
   where
-    go txOut =
+    go !txOut =
       let curMinCoin = getMinCoinTxOut pp txOut
           curCoin = txOut ^. coinTxOutL
-       in if curCoin == curMinCoin
+       in if curCoin `f` curMinCoin
```
Comment (JaredCorduan) [inline libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Out.hs:74]: sure you can't be tempted by some fun and obscure unicode symbol? :laughing:

**Review by JaredCorduan — APPROVED** [review]
looks great! I love the great docs!

File: libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Out.hs (hunk 1 above)
Comment (lehins) [reply libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Out.hs:74]: I don't think it is possible to have infix operators as function argument names .

Comment (JaredCorduan) [reply libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Out.hs:74]: of well, `f` is a great name too :grin:

**Pull request merged**